surfactant/infoextractors/file_decompression.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2025 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+# Copyright 2025 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+import tarfile
+import tempfile
+import zipfile
+from queue import Queue
+from typing import Any, Dict, Optional
+
+from loguru import logger
+
+import surfactant.plugin
+from surfactant import ContextEntry
+from surfactant.sbomtypes import SBOM, Software
+
+
+def is_compressed(filename):
+    # Map file extensions to their compression formats
+    compression_formats = {
+        ".zip": "zip",
+        ".tar": "tar",
+        ".tar.gz": "tar.gz",
+        ".tar.bz2": "tar.bz2",
+        ".tar.xz": "tar.xz",
+    }
+    for ext, format in compression_formats.items():
+        if filename.endswith(ext):
+            return format
+    return None
+
+
+@surfactant.plugin.hookimpl
+def extract_file_info(
+    sbom: SBOM, software: Software, filename: str, filetype: str, context: Queue[ContextEntry]
+) -> Optional[Dict[str, Any]]:
+    # Check if the file is compressed and get its format
+    compression_format = is_compressed(filename)
+    if not compression_format:
+        return None
+
+    # Decompress the file based on its format
+    temp_folder = check_compression_type(filename, compression_format)
+
+    # Add a new ContextEntry for the temp dir
+    new_entry = ContextEntry(
+        archive=filename, installPrefix="", extractPaths=[temp_folder], skipProcessingArchive=True
+    )
+
+    # Add new ContextEntry to queue
+    context.put(new_entry)
+    logger.info(f"New ContextEntry added for extracted files: {temp_folder}")
```

Comment:
```suggestion
    logger.info(f"New ContextEntry added for extracted files: {temp_folder}")
    return None
```

Fix pylint warning: `surfactant/infoextractors/file_decompression.py:38:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)`